### PR TITLE
Connectivity Check

### PR DESCRIPTION
**Pull request by izzyblues**
This PR implements the following changes:

- ConnectivityCheck method to reuse code for the consumer heathcheck.
- Improved godoc

**NB: THIS CHANGES ARE GOING TO BRAKE BACK COMPATIBILITY**

### CODE REVIEW

File: producer/producer.go [hunk 1]
```diff
@@ -68,13 +71,15 @@ func NewMessageProducerWithHTTPClient(config MessageProducerConfig, httpClient *
 	return &DefaultMessageProducer{config, httpClient}
 }
 
+// SendMessage is the producer method that takes care of sending a message on the queue
```
Comment (kghatton) [inline producer/producer.go:84]: Would the GoDoc not be better on the interface rather than the implementation?

**Review by kghatton — APPROVED** [review]

File: producer/producer.go (hunk 1 above)
Comment (izzyblues) [reply producer/producer.go:84]: Sounds good to me.

**Review by peteclark-ft — APPROVED** [review]

File: circle.yml [hunk 2]
```diff
@@ -1,2 +1,13 @@
...
+dependencies:
+  pre:
+    - go get github.com/axw/gocov/gocov; go get github.com/matm/gocov-html; go get -u github.com/jstemmer/go-junit-report
+  override:
+    - go get -t -d -v ./...
+    - go build -v ./...
+test:
+  override:
+    - mkdir -p $CIRCLE_TEST_REPORTS/golang
+    - go test -race -v | go-junit-report > $CIRCLE_TEST_REPORTS/golang/junit.xml
```
Comment (Nesze) [inline circle.yml]: Same as for the gonsumer, should specify the `./producer` folder.

File: producer/producer.go [hunk 3]
```diff
@@ -172,3 +187,59 @@ func envelopeMessage(key string, message string) (string, error) {
...
+func (p *DefaultMessageProducer) ConnectivityCheck() (string, error) {
+	err := p.checkMessageQueueProxyReachable()
+	if err == nil {
+		return "Connectivity to producer proxy is OK.", nil
+	}
+	log.Printf("ERROR - Producer Connectivity Check - %s", err)
+	return "Error connecting to producer proxy", err
+}
+
+func (p *DefaultMessageProducer) checkMessageQueueProxyReachable() error {
```
Comment (Nesze) [inline producer/producer.go:201]: Same as for the gonsumer, might be worth renaming this, or not treating the absence of the configured topic as an error, but just log it as WARN.

**Pull request merged**